ibis/expr/analytics.py

```diff
@@ -13,106 +13,78 @@
 # limitations under the License.
 
 
+import ibis.expr.rules as rlz
 import ibis.expr.datatypes as dt
-import ibis.expr.types as ir
-import ibis.expr.rules as rules
 import ibis.expr.operations as ops
 
+from ibis.expr.signature import Argument as Arg
 
-def _validate_closed(closed):
-    closed = closed.lower()
-    if closed not in {'left', 'right'}:
-        raise ValueError("closed must be 'left' or 'right'")
-    return closed
 
-
-class BucketLike(ir.ValueOp):
+class BucketLike(ops.ValueOp):
 
     @property
     def nbuckets(self):
         return None
 
     def output_type(self):
-        ctype = dt.Category(self.nbuckets)
-        return ctype.array_type()
+        dtype = dt.Category(self.nbuckets)
+        return dtype.array_type()
 
 
 class Bucket(BucketLike):
-
-    def __init__(self, arg, buckets, closed='left', close_extreme=True,
-                 include_under=False, include_over=False):
-        self.arg = arg
-        self.buckets = buckets
-        self.closed = _validate_closed(closed)
-
-        self.close_extreme = bool(close_extreme)
-        self.include_over = bool(include_over)
-        self.include_under = bool(include_under)
-
-        if not len(buckets):
+    arg = Arg(rlz.noop)
```

Comment:
If I understand correctly, `Arg(rlz.noop)` is equivalent to (the old) `rules.value`?

Reply:
No, noop is `toolz.identity`, just a placeholder where no rule was defined previously.

Reply:
I see. LGTM.